scipy/sparse/csgraph/_shortest_path.pyx

```diff
@@ -1510,12 +1505,8 @@ cdef void remove(FibonacciNode* node):
     # Assumptions: - node is a valid pointer
     if node.parent:
         node.parent.rank -= 1
-        if node.left_sibling:
-            node.parent.children = node.left_sibling
-        elif node.right_sibling:
+        if node.parent.children == node:
```

Comment:
I think `children` better be rename `leftmost_child` for the new representation.
```suggestion
        if node.parent.children == node:  # node is the leftmost sibling.
```